extensions/pkg/terraformer/terraformer.go

```diff
@@ -144,118 +147,141 @@ func (t *terraformer) Destroy(ctx context.Context) error {
+			return retry.MinorError(fmt.Errorf("%d/%d Terraform resources are missing", numberOfConfigResources-numberOfExistingResources, numberOfConfigResources))
 		}
 	}); err != nil {
 		return err
 	}
-	if !execute {
+	if !allConfigurationResourcesExist {
 		return nil
 	}
 
+	// Check if an existing Terraformer pod is still running. If yes, then adopt it. If no, then deploy a new pod (if
+	// necessary).
+	var (
+		pod          *corev1.Pod
+		deployNewPod = true
+	)
+
+	podList, err := t.listPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case len(podList.Items) == 1:
+		if cmd := getTerraformerCommand(&podList.Items[0]); cmd == command {
+			// adopt still existing pod
+			pod = &podList.Items[0]
+			deployNewPod = false
+		} else {
+			// delete still existing pod and wait until it's gone
+			logger.Info(fmt.Sprintf("still existing Terraform pod with command %q found - ensuring the cleanup before starting a new Terraform pod with command %q", cmd, command))
+			if err := t.EnsureCleanedUp(ctx); err != nil {
+				return err
+			}
+		}
+
+	case len(podList.Items) > 1:
+		// unreachable
+		logger.Error(fmt.Errorf("too many Terraformer pods"), "unexpected number of still existing Terraformer pods: %d", len(podList.Items))
+		if err := t.EnsureCleanedUp(ctx); err != nil {
+			return err
+		}
+	}
+
 	// In case of command == 'destroy', we need to first check whether the Terraform state contains
 	// something at all. If it does not contain anything, then the 'apply' could never be executed, probably
 	// because of syntax errors. In this case, we want to skip the Terraform destroy pod (as it wouldn't do anything
 	// anyway) and just delete the related ConfigMaps/Secrets.
 	if command == "destroy" {
-		skipApplyOrDestroyPod = t.IsStateEmpty(ctx)
+		deployNewPod = !t.IsStateEmpty(ctx)
 	}
 
-	if !skipApplyOrDestroyPod {
-		// TODO: remove after several releases
-		// ensure ownerRef for already existing state configmaps
-		if err := t.ensureStateHasOwnerRef(ctx); err != nil {
-			return fmt.Errorf("failed to ensure owner reference for the state configmap: %w", err)
-		}
-
+	if deployNewPod {
 		// Create Terraform Pod which executes the provided command
 		generateName := t.computePodGenerateName(command)
 
-		logger.Info("Deploying Terraformer Pod", "generateName", generateName)
-		pod, err := t.deployTerraformerPod(ctx, generateName, command)
+		logger.Info("Deploying Terraformer pod", "generateName", generateName)
+		pod, err = t.deployTerraformerPod(ctx, generateName, command)
 		if err != nil {
-			return fmt.Errorf("failed to deploy the Terraformer Pod with .meta.generateName %q: %w", generateName, err)
+			return fmt.Errorf("failed to deploy the Terraformer pod with .meta.generateName %q: %w", generateName, err)
 		}
+		logger.Info("Successfully created Terraformer pod", "pod", kutil.KeyFromObject(pod))
+	}
 
-		logger.Info("Successfully created Terraformer Pod", "pod", kutils.KeyFromObject(pod))
+	if pod != nil {
```

Comment:
Can this condition even be `false`?

Reply:
It can be nil if there is no running pod and the if state is already empty.